### PR TITLE
feat: using smallvec

### LINKED ISSUE #6690: Use smallvec for txs_by_sender
### Describe the feature

we know

https://github.com/paradigmxyz/reth/blob/a5dc8d66799412d3e19f6c80e4857ffac0269a5a/crates/transaction-pool/src/pool/parked.rs#L90-L90

will return fewer entries than:

https://github.com/paradigmxyz/reth/blob/a5dc8d66799412d3e19f6c80e4857ffac0269a5a/crates/transaction-pool/src/config.rs#L5-L5

this is frequently allocated+deallocated in this loop

https://github.com/paradigmxyz/reth/blob/a5dc8d66799412d3e19f6c80e4857ffac0269a5a/crates/transaction-pool/src/pool/parked.rs#L163-L163

TransactionId is 2 u64

we can move this inline to `SmallVec<[;16]>`

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by loocapro**
Closes https://github.com/paradigmxyz/reth/issues/6690

### CODE REVIEW

**Review by DaniPopes — CHANGES_REQUESTED** [review]
FYI use [GitHub keywords](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests) in the PR description to automatically close related issues once it's merged.

File: crates/transaction-pool/src/pool/parked.rs [hunk 1]
```diff
@@ -195,7 +196,10 @@ impl<T: ParkedOrd> ParkedPool<T> {
         {
             // NOTE: This will not panic due to `!last_sender_transaction.is_empty()`
             let sender_id = self.last_sender_submission.last().expect("not empty").sender_id;
-            let list = self.get_txs_by_sender(sender_id);
+            // Utilize SmallVec to efficiently handle up to 16 transactions
+            // per sender, adhering to the TXPOOL_MAX_ACCOUNT_SLOTS_PER_SENDER limit.
+            let list: SmallVec<[TransactionId; TXPOOL_MAX_ACCOUNT_SLOTS_PER_SENDER]> =
+                self.get_txs_by_sender(sender_id).into();
```
Comment (DaniPopes) [inline crates/transaction-pool/src/pool/parked.rs:202]: Actually here it doesn't make any sense, `get_txs_by_sender` already returns a Vec and we just iterate over it immediatly after. Should this be changed in the `get_txs_by_sender` return type? @mattsse

File: crates/transaction-pool/Cargo.toml [hunk 2]
```diff
@@ -48,6 +48,7 @@ serde = { workspace = true, features = ["derive", "rc"], optional = true }
 fnv = "1.0.7"
 bitflags.workspace = true
 auto_impl = "1.0"
+smallvec = "1.13.1"
```
Comment (mattsse) [inline crates/transaction-pool/Cargo.toml:51]: we can move this to workspace

File: crates/transaction-pool/src/pool/parked.rs (hunk 1 above)
Comment (mattsse) [reply crates/transaction-pool/src/pool/parked.rs:202]: yeah this should be the return type of the function

**Review by mattsse — CHANGES_REQUESTED** [review]

File: Cargo.toml [hunk 3]
```diff
@@ -224,6 +224,7 @@ hex-literal = "0.4"
 once_cell = "1.17"
 syn = "2.0"
 nybbles = "0.1"
+smallvec = "1.13.1"
```
Comment (emhane) [inline Cargo.toml:227]:
```suggestion
smallvec = "1.13"
```

**Review by emhane — APPROVED** [review]
nice to see `smallvec` in the codebase. we can use this type in a lot more places for small performance improvements.

**Review by DaniPopes — APPROVED** [review]

**Review by mattsse — APPROVED** [review]

**Pull request merged**